### PR TITLE
Introduces page_size to pagination navigation

### PR DESCRIPTION
**Pull request by hennevogel**
Use the kaminari per scope instead of a hand crafted show_more parameter.

Turns this

image

into this

image

### CODE REVIEW

File: src/api/app/views/webui/users/notifications/_page_size_navigation.html.haml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- if paginated_objects.total_pages > 1
+  .text-center
+    .btn-group{ role: 'group', 'aria-label': 'Page Size' }
+      - page_size = params[:page_size]&.to_i || paginated_objects.default_per_page
+      - [paginated_objects.default_per_page, 100, 300].each do |size|
+        - active = 'active' if page_size == size
+        = link_to(url_for( page_size: size ), class: "btn btn-sm btn-outline-info #{active}") do
+          = size
+    \/ page
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/views/webui/users/notifications/_page_size_navigation.html.haml:9]: Would it? Why?

[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/views/webui/users/notifications/_page_size_navigation.html.haml]: pushed

File: src/api/lib/tasks/dev/notifications.rake [hunk 2]
```diff
@@ -21,7 +19,8 @@ namespace :dev do
       requestor_project = Project.find_by(name: 'requestor_project') || RakeSupport.create_and_assign_project('requestor_project', requestor)
 
       # Create notification for roles revoked
-      User.find_by(login: 'Iggy').run_as do
+      iggy = User.where(login: 'Iggy').first || create(:confirmed_user, login: 'Iggy', create_home_project: true)
+      iggy.run_as do
```
Comment (eduardoj) [inline src/api/lib/tasks/dev/notifications.rake:23]: That works on a fresh database.

But for a database where the `dev:test_data:create` has been run, the home project of the user Iggy is not created. See: https://github.com/openSUSE/open-build-service/blob/5a23b91410ef70ecf06e0aebc20f99e950f93791/src/api/lib/tasks/dev.rake#L103

We can create a new user or modify the creation of the the Iggy user to add `create_home_project: true` where I pointed to before.

Comment (eduardoj) [reply src/api/lib/tasks/dev/notifications.rake:23]: The home project for the user Iggy gets created later on, here: https://github.com/openSUSE/open-build-service/blob/5a23b91410ef70ecf06e0aebc20f99e950f93791/src/api/lib/tasks/dev.rake#L120

No need to change the code as I suggested. Sorry for the noise.

File: src/api/lib/tasks/dev/notifications.rake [hunk 3]
```diff
@@ -21,7 +19,8 @@ namespace :dev do
       requestor_project = Project.find_by(name: 'requestor_project') || RakeSupport.create_and_assign_project('requestor_project', requestor)
 
       # Create notification for roles revoked
-      User.find_by(login: 'Iggy').run_as do
+      iggy = User.where(login: 'Iggy').first || create(:confirmed_user, login: 'Iggy', create_home_project: true)
```
Comment (eduardoj) [inline src/api/lib/tasks/dev/notifications.rake:22]:
```suggestion
      iggy = User.find_by(login: 'Iggy') || create(:confirmed_user, login: 'Iggy', :with_home)
```
Using prevents from `find_by` query with `where` and then apply `first`.

Also, make use of the ` with_home` defined trait.

Comment (hennevogel) [reply src/api/lib/tasks/dev/notifications.rake]: done

File: src/api/lib/tasks/dev/notifications.rake [hunk 4]
```diff
@@ -21,7 +19,8 @@ namespace :dev do
       requestor_project = Project.find_by(name: 'requestor_project') || RakeSupport.create_and_assign_project('requestor_project', requestor)
 
       # Create notification for roles revoked
-      User.find_by(login: 'Iggy').run_as do
+      iggy = User.find_by(login: 'Iggy') || create(:confirmed_user, login: 'Iggy', :with_home)
```
Comment (eduardoj) [inline src/api/lib/tasks/dev/notifications.rake:22]:
```suggestion
      iggy = User.find_by(login: 'Iggy') || create(:confirmed_user, :with_home, login: 'Iggy')
```
Sorry, I meant this way.

**Review by eduardoj — APPROVED** [review]

**Pull request merged**